config/module_input_intentions_test.go

```diff
@@ -0,0 +1,277 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIntentionsModuleInputConfig_Copy(t *testing.T) {
+	t.Parallel()
+
+	finalizedConf := &IntentionsModuleInputConfig{}
+	finalizedConf.Finalize()
+
+	cases := []struct {
+		name string
+		a    *IntentionsModuleInputConfig
+	}{
+		{
+			"nil",
+			nil,
+		},
+		{
+			"empty",
+			&IntentionsModuleInputConfig{},
+		},
+		{
+			"finalized",
+			finalizedConf,
+		},
+		{
+			"happy_path",
+			&IntentionsModuleInputConfig{
+				IntentionsMonitorConfig{
+					Datacenter: String("dc"),
+					Namespace:  String("namespace"),
+					SourceServices: &IntentionsServicesConfig{
+						Regexp: String("^web.*"),
+					},
+					DestinationServices: &IntentionsServicesConfig{
+						Regexp: String("^api.*"),
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			r := tc.a.Copy()
+			if tc.a == nil {
```

Comment:
Can you please elaborate on this?

Reply:
An interface only equals nil if both the type and value are nil therefore we can not do an assert.Nil(t, tc.a, r). Interface nil type is not equal to tc.a so first we must check tc.a equals nil and then we can assert nil for t and r.